tests/test_cancel_event.py

```diff
@@ -0,0 +1,278 @@
+    doesn't correspond to an event in the database
+    """
+    return response.status_code == 404
+
+
+def check_event_update_safe(old_event: event_models.Event,
+                            new_event: event_models.Event) -> bool:
+    """
+    Checks if a newly acquired event object is the same as the old, except for
+    an enum which is expected to now be cancelled on the new object.
+    """
+    old_event.status = event_models.EventStatusEnum.cancelled
+    return old_event == new_event
+
+
+def get_event_cancel_response(event: event_models.Event,
+                              header_dict: Dict[str, Any]) -> HTTPResponse:
+    """
+    Makes the endpoint call with the given event
+    and header dict, returning the response.
+    """
+    json_payload = get_valid_call_payload(event)
+    endpoint_url = get_cancel_event_endpoint_url()
+    response = client.patch(endpoint_url,
+                            json=json_payload,
+                            headers=header_dict)
+    return response
+
+
+def event_in_database(event: event_models.Event) -> bool:
+    """
+    Checks if the given event is in the database
+    """
+    id_dict = async_to_sync(event_utils.generate_event_id_dict)(event)
+    var = event_utils.events_collection().find_one(id_dict)
+    return bool(var)
+
+
+class TestCancelEvent:
+    def test_creator_cancel_event(self, registered_user: user_models.User,
+                                  registered_active_event_factory: Callable[
+                                      [], event_models.Event],
+                                  get_header_dict_from_user: Callable[
+                                      [user_models.User], Dict[str, Any]]
+                                  ):
+        """
+        Tries to call the cancel endpoint from a User
+        that created an event, expecting success
+        """
+        event = registered_active_event_factory(
+            registered_user)  # todo: figure out why this is unexpected argument
+        header_dict = get_header_dict_from_user(registered_user)
+
+        user_identifier = user_models.UserIdentifier(user_id=registered_user.id)
+        old_user_data = async_to_sync(
+            user_utils.get_user_info_by_identifier)(
+            user_identifier)  # todo: check this vs asyncio, then normalize
+        old_event_data = asyncio.run(
+            event_utils.get_event_by_id(event_id=event.id))
+
+        response = get_event_cancel_response(event, header_dict)
+
+        # get new state from database for comparison
+        new_user_data = asyncio.run(
+            user_utils.get_user_info_by_identifier(user_identifier))
+        new_event_data = asyncio.run(
+            event_utils.get_event_by_id(event_id=event.id))
+
+        assert check_valid_cancel_response(response)
+        assert new_user_data == old_user_data
+
+        # pylint: disable=no-member
+        #   -this is necessary to pass tests
```

Comment:
Is this ok to disable?

Reply:
not really, should be disabling this _at most_ on a single line at a time.